esmvalcore/preprocessor/_regrid.py

```diff
@@ -600,6 +598,13 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
     if not isinstance(target_grid, iris.cube.Cube):
         raise ValueError('Expecting a cube, got {}.'.format(target_grid))
 
+    if isinstance(scheme_args, list):
+        if 'src_cube' in scheme_args:
+            scheme['src_cube'] = cube
+        if 'grid_cube' in scheme_args:
+            scheme['grid_cube'] = target_grid
+        loaded_scheme = obj(**scheme)
+
```

Comment:
Yes I thought it would fit better in those lines as well. However, there two schemes in [iris-esmf](https://iris-esmf-regrid.readthedocs.io/en/latest/_api_generated/esmf_regrid.schemes.html). 

The current main, supports the call to `esmf_regrid.schemes.ESMFAreaWeighted(mdtol=0)`, which returns a scheme to be called by cube.regrid.

But `esmf_regrid.schemes.regrid_rectilinear_to_rectilinear` does not return a scheme, it's the regridding routine on it's own. It expects cubes as inputs and returns a regridded cube as output. And at that point the target cube has not been generated yet. 

That is why the arguments are being checked. Because one scheme does not require them (unless we want to allow mdtol to be set as well, which is not possible right now), whereas another one fails if the src_cube and grid_cube are not given as inputs.

Would it work for you if the routine checked first the `if isinstance(target_grid, str)` block and then the `if isinstance(scheme, dict)` one? At first sight it does not look like it would break anything and it would be more organised.

Reply:
Ah yes, you are totally right, I didn't notice that `target_grid` is defined only later! I think swapping the blocks `if isinstance(target_grid, str)` and `if isinstance(scheme, dict)` is a good idea!

About the `mdtol`: I think it currently **is** possible to set this with

```yaml
scheme:
  reference: esmf_regrid.schemes.ESMFAreaWeighted
  mdtol: 0.5
```